### PR TITLE
Build PCI Root Bridge Resource Info Hob

### PR DESCRIPTION
**Pull request by myguitar**
This will create a HOB for PCI Root Bridge Resource information.
This HOB can be used to update resource ranges of multiple root bridges for
platform PCI tree ASL.
A payload can also use this info to skip duplicated root bridge scan.

Signed-off-by: Aiden Park <aiden.park@intel.com>

### CODE REVIEW

File: BootloaderCommonPkg/Include/Library/BootloaderCommonLib.h [hunk 1]
```diff
@@ -61,6 +61,19 @@ typedef struct {
...
+  PciBarTypeIo16,
+  PciBarTypeIo32,
+  PciBarTypeMem32,
+  PciBarTypePMem32,
+  PciBarTypeMem64,
+  PciBarTypePMem64,
+  PciBarTypeIo,
+  PciBarTypeMem,
+  PciBarTypeMaxType
+} PCI_BAR_TYPE;
```
Comment (mauricema) [inline BootloaderCommonPkg/Include/Library/BootloaderCommonLib.h:75]: Should we move them into PciRootBridgeInfoGuid.h ?

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c [hunk 2]
```diff
@@ -1162,6 +1164,43 @@ DumpPciResources (
     CurrentLink = CurrentLink->ForwardLink;
   }
 }
+
+/**
+  Dump PCI Info HOBs
+
+**/
+VOID
+DumpPciInfoHob (
```
Comment (mauricema) [inline BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c:1173]: DumpPciRootBridgeInfoHob ?

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c [hunk 3]
```diff
@@ -1291,17 +1334,128 @@ PciScanRootBridges (
...
+  @param [in] RootBridge        A pointer of Root Bridges List
+  @param [in] RootBridgeCount   The number of found root bridges
+
+  @retval EFI_SUCCESS           Create HOBs successfully
+  @retval EFI_OUT_OF_RESOURCES  Out of Hob resource
+  @retval EFI_LOAD_ERROR        Invalid data integrity
+
+ **/
+EFI_STATUS
+BuildPciInfoHob (
```
Comment (mauricema) [inline BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c:1414]: BuildPciRootBridgeHob

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c [hunk 4]
```diff
@@ -1291,17 +1334,128 @@ PciScanRootBridges (
...
+  @param [in] RootBridgeInfoHob   A pointer of root bridge resource info HOB
+  @param [in] MaxRootBridgeCount  The maximum number of root bridges
+
+  @retval EFI_SUCCESS             Create HOBs successfully
+  @retval EFI_INVALID_PARAMETER   Invalid parameters passing
+  @retval EFI_LOAD_ERROR          Invalid data integrity
+
+ **/
+EFI_STATUS
+BuildPciRootBridgeInfoHob (
```
Comment (mauricema) [inline BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c:1363]: FillPciRootBridgeInfo ?

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c [hunk 5]
```diff
@@ -1291,17 +1334,128 @@ PciScanRootBridges (
...
+    }
+
+    CurrentLink = CurrentLink->ForwardLink;
+  }
+
+  if (RootBridgeInfoHob->Count != RootBridgeCount) {
+    Status = EFI_LOAD_ERROR;
+  }
+
+  if (EFI_ERROR (Status)) {
```
Comment (mauricema) [inline BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c:1453]: Maybe rollback Count to 0 is good enough?   EX:  You might still want the Revision info.

File: BootloaderCommonPkg/Include/Library/BootloaderCommonLib.h (hunk 1 above)
Comment (myguitar) [reply BootloaderCommonPkg/Include/Library/BootloaderCommonLib.h]: Moved to PciRootBridgeInfoGuid.h.

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c (hunk 2 above)
Comment (myguitar) [reply BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c]: Renamed to DumpPciRootBridgeInfoHob

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c (hunk 3 above)
Comment (myguitar) [reply BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c]: Renamed to BuildPciRootBridgeInfoHob

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c (hunk 4 above)
Comment (myguitar) [reply BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c]: Renamed to FillPciRootBridgeInfo

File: BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c (hunk 5 above)
Comment (myguitar) [reply BootloaderCorePkg/Library/PciEnumerationLib/PciEnumerationLib.c:1451]: Updated as commented. Thanks.

**Pull request merged**